### PR TITLE
Migrate `rlib-format-packed-bundled-libs-2`, `native-link-modifier-whole-archive` and `no-builtins-attribute` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Please try:

try-job: x86_64-msvc
try-job: test-various
try-job: armhf-gnu
try-job: aarch64-apple
try-job: x86_64-gnu-llvm-18

### CODE REVIEW

File: tests/run-make/rlib-format-packed-bundled-libs-2/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use run_make_support::{llvm_ar, llvm_readobj, regex, rfs, rust_lib_name, rustc};
+
+fn main() {
+    // Build a strangely named dependency.
+    rustc().input("native_dep.rs").crate_type("staticlib").output("native_dep.ext").run();
+
+    rustc().input("rust_dep.rs").crate_type("rlib").arg("-Zpacked_bundled_libs").run();
+    let symbols = llvm_readobj().symbols().input(rust_lib_name("rust_dep")).run().stdout_utf8();
+    let re = regex::Regex::new("U.*native_f1").unwrap();
+    assert!(re.is_match(&symbols));
```
Comment (jieyouxu) [inline tests/run-make/rlib-format-packed-bundled-libs-2/rmake.rs:18]: Suggestion: as discussed,more verbose logging about *what* failed.

Comment (Oneirical) [reply tests/run-make/rlib-format-packed-bundled-libs-2/rmake.rs:18]: Reminder for myself: `assert_stdout_contains_regex` and `llvm_nm`

Comment (Oneirical) [reply tests/run-make/rlib-format-packed-bundled-libs-2/rmake.rs]: `assert_stdout_contains_regex` implemented.

**Comment by Oneirical** [comment]
Looks like the symbol does appear... under a different format than Linux. Let's see if `llvm_nm` is any better. @bors try

**Review by jieyouxu — APPROVED** [review]
Thanks, r=me if the additional round of try jobs come back green.

**Comment by Mark-Simulacrum** [comment]
@rust-timer build 83dcdb3a5dad0ed1e3e1fadc848d3f7727b41aa5

**Pull request merged**